x-pack/plugin/build.gradle

```diff
@@ -98,6 +98,26 @@ tasks.named("yamlRestTestV7CompatTransform").configure{ task ->
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given default", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given top_n is 2", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given top_n is 3", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given top_n is greater than the job count", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given overall_score filter", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given exclude_interim", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given string start and end params", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given epoch start and end params", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given invalid start param", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given invalid end param", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given bucket_span", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given bucket_span and top_n is 2", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given bucket_span and overall_score filter", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given no matching jobs", "behaviour change #81377 - setting default value of 30 days for model prune window")
+  task.skipTest("ml/jobs_get_result_overall_buckets/Test overall buckets given bucket_span is smaller than max job bucket_span", "behaviour change #81377 - setting default value of 30 days for model prune window")
```

Comment:
It's pretty bad for us to be losing all our overall buckets REST testing because of a new default, when this new default shouldn't impact the overall buckets functionality at all.

I think a better solution is to open a separate PR that explicitly sets the model prune window in the overall buckets REST tests.  That PR can then be backported to 8.0 and 7.16.  Once those backports are merged these skips can be removed.

Or if I've guessed wrongly about what the underlying problem is here and you don't think that would work, why does setting a default prune window break the overall buckets functionality?

Reply:
Setting a default value of 30 days breaks these tests because they incorporate a job that has a bucket span of 17 minutes and we have a validation check that the `model_prune_window` value must be an integral multiple of the bucket span.

As you suggest the optimal solution is to backport a change to the tests that explicitly sets a compatible value for the `model_prune_window` then re-instate these tests.

Reply:
Oh, interesting.  I think this reveals a problem with the new default then.  It's not a good UX if someone submits a perfectly good job config and we add a default to it that's incompatible.

I think this PR should be changed to set the default to the closest multiple of the bucket span that's greater than or equal to 30 days.  Generally that will be 30 days, but for weird bucket spans it might be 30 days and 4 minutes or something.

Reply:
Sure, although it may make messaging a little awkward ;-)

Reply:
> set the default to the closest multiple of the bucket span that's greater than or equal to 30 days

In fact, since we're introducing a formula we should future proof it in other ways too.  With a 1 week bucket span we'd be pruning models after 5 buckets.  I think we should keep models for at least 20 buckets, so:

```
model_prune_window = max(20 * bucketSpan, closest multiple of the bucket span that's greater than or equal to 30 days)
```

Reply:
> it may make messaging a little awkward

I think it's OK to message it as the greater of 30 days or 20 times bucket span.  The slight rounding for weird bucket spans can be an implementation detail, very unlikely to be seen by sensible users.